File: packages/types/src/page-collection/page-collection.ts
/**
 * SPDX-FileCopyrightText: 2023-present Plane Software, Inc.
 * SPDX-License-Identifier: LicenseRef-Plane-Commercial
 *
 * Licensed under the Plane Commercial License (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * https://plane.so/legals/eula
 *
 * DO NOT remove or modify this notice.
 * NOTICE: Proprietary and confidential. Unauthorized use or distribution is prohibited.
 */

import type { TPaginationInfo } from "../common";
import type { TPage, TPageFilterProps, TSubPageDetails } from "../page";

export type TPageCollection = {
  id: string;
  collection: string;
  page: string;
  sort_order?: number | null;
  workspace?: string;
  created_at?: Date;
  updated_at?: Date;
  created_by?: string;
  updated_by?: string;
};

type TPageCollectionMetadata = Pick<
  TPageCollection,
  "workspace" | "created_at" | "updated_at" | "created_by" | "updated_by"
>;

export type TPageCollectionBranchParams = {
  parent_id?: string | null;
  search?: string;
  filters?: TPageFilterProps;
  cursor?: string;
  per_page?: number;
};

type TCollectionBranchPageSummaryMetadata = Pick<
  TPage,
  | "workspace"
  | "sub_pages_count"
  | "is_shared"
  | "shared_access"
  | "owned_by"
  | "deleted_at"
  | "is_description_empty"
  | "updated_at"
  | "updated_by"
  | "moved_to_page"
  | "moved_to_project"
  | "sort_order"
  | "created_at"
  | "created_by"
  | "is_favorite"
>;

export type TCollectionBranchPageSummary = TSubPageDetails &
  TCollectionBranchPageSummaryMetadata &
  Required<Pick<TPage, "collection_id">>;

export type TCollectionBranchRow = TPageCollectionMetadata & {
  page_collection_id?: string | null;
  collection_id: string;
  page: TCollectionBranchPageSummary;
  sort_order?: number | null;
  parent_id?: string | null;
};

export type TPageCollectionBranchResponse = TPaginationInfo & {
  results: TCollectionBranchRow[];
};

export type TCollectionAddablePage = Pick<TPage, "id" | "name" | "logo_props">;

export type TPageCollectionCreatePayload = {
  page_ids: string[];
  sort_orders?: Record<string, number>;
};

export type TPageCollectionUpdatePayload = {
  collection?: string;
  sort_order?: number;
};

File: apps/api/plane/utils/filters/page_collection.py
# SPDX-FileCopyrightText: 2023-present Plane Software, Inc.
# SPDX-License-Identifier: LicenseRef-Plane-Commercial
#
# Licensed under the Plane Commercial License (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
# https://plane.so/legals/eula
#
# DO NOT remove or modify this notice.
# NOTICE: Proprietary and confidential. Unauthorized use or distribution is prohibited.

from django_filters import filters

from plane.db.models import Page
from plane.utils.filters.filterset import BaseFilterSet, UUIDInFilter


class CollectionPageFilterSet(BaseFilterSet):
    created_by = UUIDInFilter(field_name="created_by_id", lookup_expr="in")
    favorites = filters.BooleanFilter(field_name="is_favorite")
    created_at__gte = filters.CharFilter(field_name="created_at", method="filter_datetime_date_gte")
    created_at__lte = filters.CharFilter(field_name="created_at", method="filter_datetime_date_lte")

    class Meta:
        model = Page
        fields = []
